services/workflows-service/src/workflow/cron/ongoing-monitoring.cron.ts

```diff
@@ -0,0 +1,218 @@
+import { WorkflowDefinitionService } from '@/workflow-defintion/workflow-definition.service';
+import { BusinessReportService } from '@/business-report/business-report.service';
+import { TProjectIds } from '@/types';
+import { ajv } from '@/common/ajv/ajv.validator';
+import { ValidationError } from '@/errors';
+
+@Injectable()
+export class OngoingMonitoringCron {
+  private readonly lockKey = ONGOING_MONITORING_LOCK_KEY;
+
+  constructor(
+    protected readonly prisma: PrismaService,
+    protected readonly logger: AppLoggerService,
+    protected readonly workflowService: WorkflowService,
+    protected readonly workflowDefinitionService: WorkflowDefinitionService,
+    protected readonly customerService: CustomerService,
+    protected readonly businessService: BusinessService,
+    protected readonly businessReportService: BusinessReportService,
+  ) {}
+
+  private readonly PENDING_ONGOING_MONITORING_STATE = ['pending-ongoing-monitoring'];
+
+  @Cron(CronExpression.EVERY_DAY_AT_MIDNIGHT)
+  async handleCron() {
+    const lockAcquired = await this.prisma.acquireLock(this.lockKey);
+
+    if (!lockAcquired) {
+      this.logger.warn('Lock not acquired, another instance might be running the job.');
+
+      return;
+    }
+
+    try {
+      const customers = await this.customerService.list({
+        select: { projects: true, features: true, definitionConfigs: true, id: true },
+      });
+
+      const processConfiguration = await this.fetchCustomerProcessConfiguration(customers);
+
+      for (const {
+        projectIds,
+        workflowDefinition,
+        definitionConfig: customerDefinitionConfig,
+      } of processConfiguration) {
+        const businesses = await this.businessService.list({}, projectIds);
+
+        for (const business of businesses) {
+          const workflowDefinitionConfig =
+            this.findDefinitionConfig(business.definitionConfigs) || customerDefinitionConfig;
+
+          if (!workflowDefinitionConfig?.active) continue;
+
+          const intervalInDays = workflowDefinitionConfig.intervalInDays;
+          const lastRunDate = (
+            business.metadata as { lastOngoingAuditReportRunDate?: number } | undefined
+          )?.lastOngoingAuditReportRunDate;
+
+          const lastReceivedReport = (
+            await this.businessReportService.findMany(
+              {
+                where: {
+                  businessId: business.id,
+                  projectId: business.projectId,
+                },
+                orderBy: {
+                  createdAt: 'desc',
+                },
+                take: 1,
+              },
+              projectIds,
+            )
+          )[0];
+
+          if (!lastReceivedReport) {
+            return this.invokeOngoingAuditReport({
+              business,
+              workflowDefinitionConfig,
+              workflowDefinitionId: workflowDefinition.id,
+              currentProjectId: business.projectId,
+              projectIds: projectIds,
```

Comment:
Should probably be: `projectId: business.projectId` to reduce unscary data access of this function